distribution/sdk/sample-cometd/cometd-java-example/src/main/java/org/codice/ddf/sdk/cometd/AsyncClient.java

```diff
@@ -0,0 +1,231 @@
+    private static final String DOWNLOAD_CONTEXT = "/services/catalog/sources/";
+    private static final String DOWNLOAD_TRANSFORM = "?transform=resource";
+
+    private static String asyncClientId;
+    private static Map<String, Object> queryResponse;
+    private static Map<String, Object> activities;
+    private static Map<String, Object> notifications;
+
+    private String url;
+    private BayeuxClient client;
+    private Map<String, Object> emptyMessage = new HashMap<>();
+
+    /**
+     * Creates a ddf async client with default channel subscriptions
+     * @param url - target ddf url
+     * @param disableValidation - disable ssl hostname validation
+     * @throws Exception
+     */
+    public AsyncClient(String url, Boolean disableValidation) throws Exception {
+
+        this.url = url;
+
+        HttpClient httpClient = new HttpClient();
+
+        httpClient.start();
+
+        Map<String, Object> options = new HashMap<String, Object>();
+        ClientTransport transport = new LongPollingTransport(options, httpClient);
+
+        // DisableValidation
+        if (disableValidation) {
+            doTrustAllCertificates();
+        }
+
+        this.client = new BayeuxClient(url + COMETD_CONTEXT, transport);
+
+        client.handshake(new ClientSessionChannel.MessageListener() {
+            public void onMessage(ClientSessionChannel channel, Message message) {
+                if (message.isSuccessful()) {
+                    asyncClientId = channel.getSession().getId();
+
+                    client.getChannel(ALL_ACTIVITIES).addListener(new ClientSessionChannel.MessageListener() {
+                        public void onMessage(ClientSessionChannel channel, Message message) {
+                            activities = message.getDataAsMap();
+                        }
+                    });
+
+                    client.getChannel(ALL_NOTIFICATIONS).addListener(new ClientSessionChannel.MessageListener() {
+                        public void onMessage(ClientSessionChannel channel, Message message) {
+                            notifications = message.getDataAsMap();
+                        }
+                    });
+                }
+            }
+        });
+        boolean handshaken = client.waitFor(1000, BayeuxClient.State.CONNECTED);
+
+        checkAllActivities();
+        checkAllNotifications();
+    }
+
+    /**
+     * Get all notification messages
+     * @return - notifications map
+     */
+    public Map<String, Object> getNotifications() {
+        return notifications;
+    }
+
+    /**
+     * Get all activities messages
+     * @return - activities map
+     */
+    public Map<String, Object> getActivities() {
+        return activities;
+    }
+
+    /**
+     * Get all query response messages
+     * @return - query response map
+     */
+    public Map<String, Object> getQueryResponse() {
+        return queryResponse;
+    }
+
+    /**
+     * Retrieves all persisted notifications
```

Comment:
We may want to explain here why we publish an empty message to get the latest stuff and also mention where those messages will be received.